salt/config/__init__.py

```diff
@@ -944,6 +944,9 @@ def _gather_buffer_space():
     # Always generate minion id in lowercase.
     'minion_id_lowercase': bool,
 
+    # Remove this single domain (e.g. 'foo.org') in a generated minion id.
+    'minion_id_remove_domain': six.string_types,
```

Comment:
This can also have `bool`. In case of `False`, it is as same as an empty string. In case of `True`, it removes _all_ domains. This is just a suggestion.

Reply:
That would be if there are several domains. However, if someone has one domain network and wants it w/o domains at all, then why not?

Reply:
> Also, it would confuse me to set a bool to 'foo.org', and I would half expect Python to reject it

I am not following. In case of boolean, you do:

`minion_id_remove_domain = true`

And so then every ID gets domain remove all the time, regardless. That is quite useful if you have one admin network and your machines are, say, mainly hardware and you want to access them via admin network, which sits behind another internal DNS. Or it is a VM, but it is not called `files.home.lan` but some coding scheme that tells you where exactly this machine hosted.

Reply:
This sounds to me like a description of a new PR on a new variable `minion_id_remove_all_domains: bool`

Reply:
If you would create such an option exists, I would 
leave `minion_id_remove_all_domains = false` and
set `minion_id_remove_domain = foo.bar` because I only want one single domain to be removed.


Lately, [there was an idea](https://github.com/saltstack/salt/issues/48629#issuecomment-414678059) for yet another options:

    minion_white_list_domain: good1.org, good2.org, good3.org
    minion_black_list_domain: evil1.org, evil2.org, evil3.org

Reply:
@isbm In just an abstract case in Linux for `mnion_id_remove_domain = True` I'd expect it'll remove the domain specified in `resolv.conf` ;)

Reply:
@DmitryKuzmenko which is actually true (as well). This is exactly the point: clean away _any_ kind of domain, no matter where it comes from. At the end you end up only with the hostname. Pretty useful for small scale deployments.

If I have at all only 5-10 machines in a soho business office to look after, all of them are resolvable just by a name although might sit in a different domains, on what reason would I dislike such feature?

Reply:
> This sounds to me like a description of a new PR on a new variable > `minion_id_remove_all_domains: bool`

Yes, it could be. But then it gets complicated, because "all domains" is like a list of possible domains, while "remove domain" sounds just as it is: remove a domain from the ID. And if it is `"true"` (or `"*"` if you want) then it will just match everything. Why I am not in favour of an asterisk, because it will look like probably other options available, e.g. `foo*`.

Hence I would prefer to have one option that can simply take different kind of options. If it is `true`, then just razor cut everything. If it is `false`, do not touch anything. If it is a specific domain, remove only it.

_MAYBE_, recalling to the black/white list, you want to add a simple matcher by an asterisk then. But that is indeed a different topic/PR and discussion, I would refrain from going there.